### PR TITLE
Documentation improvements.

### LINKED ISSUE #67: Document Re_pcre
I have some questions like:

What does `pmatch` do?

See ocaml/opam#2217.

### PR DESCRIPTION
**Pull request by Drup**
Should fix #67 and #70
@dsheets ?

### CODE REVIEW

File: lib/re.mli [hunk 1]
```diff
@@ -225,7 +225,11 @@ val rep1 : t -> t
 (** 1 or more matches *)
 
 val repn : t -> int -> int option -> t
-(** Repeated matches *)
+(** Repeated matches.
+    [repn re i j] matches [re] at least [i] times
+    and at most [j] times, bounds included.
+    [j = None] means to upper bound.
```
Comment (rgrinberg) [inline lib/re.mli]: s/to/no/

File: lib/re.mli [hunk 2]
```diff
@@ -225,7 +225,11 @@ val rep1 : t -> t
 (** 1 or more matches *)
 
 val repn : t -> int -> int option -> t
-(** Repeated matches *)
+(** Repeated matches.
```
Comment (rgrinberg) [inline lib/re.mli]: Actually i'm not even sure what "Repeated matches" adds here.

**Comment by rgrinberg** [comment]
+1

**Comment by Drup** [comment]
Both fixed.

File: lib/re_pcre.mli [hunk 3]
```diff
@@ -12,18 +12,25 @@ type split_result =
...
+(** Equivalent to {!Re.exec}. *)
 
 val get_substring : groups -> int -> string
+(** Equivalent to {!Re.Group.get}. *)
 
 val get_substring_ofs : groups -> int -> int * int
+(** Equivalent to {!Re.Group.offset}. *)
 
 val pmatch : rex:regexp -> string -> bool
+(** Equivalent to {!Re.exec}. *)
```
Comment (dsheets) [inline lib/re_pcre.mli]: `execp`?

**Comment by dsheets** [comment]
Except for the `pmatch` typo, lgtm.

**Pull request merged**